js/reference.js

```diff
@@ -468,18 +465,48 @@ var ERMrest = (function(module) {
             }
 
             function getDefaults() {
+                // any row in data is sufficient. data should include the full set of visible columns
+                var visibleColumnsNames = Object.keys(data[0]);
+                var tableColumnsNames = self._table.columns._columns.map(function (col) {
+                    return col.name;
+                });
                 // This is gets the difference between the table's set of columns and the reference's set of columns
-                var defaults = module._columnDiff(self._table.columns._columns, self.columns);
+                var defaults = columnDiff(tableColumnsNames, visibleColumnsNames);
+
+                // loop through the data and add any columns not set to the defaults
+                visibleColumnsNames.forEach(function (columnName) {
+                    var notSet = true;
+                    /**
+                     * Loop through the rows and make sure the current column is not set in any of the rows.
+                     * If not set in any row, add it to defaults.
+                     * If 1 row has it set and none of the others, it cannot be part of defaults
+                    **/
+                    data.forEach(function (row) {
+                        if (row[columnName]) notSet = false;
```

Comment:
This `forEach` will run for all the rows in data, even if it matches the first one. 

Change it to [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) or `for` loop with `break`; so that it will return the `notSet` value as soon as it find one row with `colName`.